src/acl/agent.test.ts

```diff
@@ -848,6 +851,540 @@ describe("getAgentResourceAccessModesAll", () => {
+    );
+    expect(updatedQuads[2].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Read"
+    );
+    expect(updatedQuads[3].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#agentClass"
+    );
+    expect(updatedQuads[3].object.value).toBe(
+      "http://xmlns.com/foaf/0.1/Agent"
+    );
+  });
+
+  it("does not change ACL rules that also apply to other Agents", () => {
+    const sourceDataset = addAclRuleQuads(
+      getMockDataset("https://arbitrary.pod/resource.acl"),
+      "https://some.pod/profileDoc#webId",
+      "https://arbitrary.pod/resource",
+      { read: true, append: false, write: false, control: false },
+      "resource"
+    );
+    const aclRuleSubject = Array.from(sourceDataset)[0].subject;
+    sourceDataset.add(
+      DataFactory.quad(
+        aclRuleSubject,
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#agent"),
+        DataFactory.namedNode("https://some-other.pod/profileDoc#webId")
+      )
+    );
+
+    const updatedDataset = unstable_setAgentResourceAccessModes(
+      sourceDataset,
+      "https://some.pod/profileDoc#webId",
+      {
+        read: false,
+        append: true,
+        write: false,
+        control: false,
+      }
+    );
+
+    const updatedQuads: Quad[] = Array.from(updatedDataset);
+    expect(updatedQuads).toHaveLength(8);
+    expect(updatedQuads[0].predicate.value).toBe(
+      "http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
+    );
+    expect(updatedQuads[0].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Authorization"
+    );
+    expect(updatedQuads[1].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#accessTo"
+    );
+    expect(updatedQuads[1].object.value).toBe("https://arbitrary.pod/resource");
+    expect(updatedQuads[2].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#mode"
+    );
+    expect(updatedQuads[2].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Read"
+    );
+    expect(updatedQuads[3].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#agent"
+    );
+    expect(updatedQuads[3].object.value).toBe(
+      "https://some-other.pod/profileDoc#webId"
+    );
+    expect(updatedQuads[4].predicate.value).toBe(
+      "http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
+    );
+    expect(updatedQuads[4].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Authorization"
+    );
+    expect(updatedQuads[5].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#mode"
+    );
+    expect(updatedQuads[5].object.value).toBe(
+      "http://www.w3.org/ns/auth/acl#Append"
+    );
+    expect(updatedQuads[6].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#accessTo"
+    );
+    expect(updatedQuads[6].object.value).toBe("https://arbitrary.pod/resource");
+    expect(updatedQuads[7].predicate.value).toBe(
+      "http://www.w3.org/ns/auth/acl#agent"
+    );
+    expect(updatedQuads[7].object.value).toBe(
+      "https://some.pod/profileDoc#webId"
+    );
+  });
```

Comment:
This should also test that the subjects between `updatedQuads[2]` and `updatedQuads[5]` are different, otherwise `https://some-other.pod/profileDoc#webId` gets extended access rights as a side effect.

Reply:
Good catch, fixed in 628b1345fc6bf7549c1fa279354ea0b29a4f4581.